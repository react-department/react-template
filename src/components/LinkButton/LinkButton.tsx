import { Link } from 'react-router-dom';

import type { ReactElement } from 'react';
import type ILinkButton from './interfaces/ILinkButton';

import styles from './sass/LinkButton.module.scss';

/**
 * Primary UI component for displaying a link in the project.
 */
function LinkButton({
  to,
  title,
  onClick,
  dataTestId,
}: ILinkButton): ReactElement {
  return (
    <Link
      to={to}
      className={styles.linkButton}
      onClick={onClick}
      data-testid={dataTestId}
    >
      { title }
    </Link>
  );
}

export default LinkButton;
