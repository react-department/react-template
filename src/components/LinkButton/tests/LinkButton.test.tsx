import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import LinkButton from '../LinkButton';

describe('<LinkButton />', () => {
  test('should render component', () => {
    const { asFragment } = render(<LinkButton to="/" title="Link title" onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('check if component has title', () => {
    render(<LinkButton to="/" title="Link title" onClick={() => {}} />);
    expect(screen.getByText('Link title')).toBeInTheDocument();
  });
});
