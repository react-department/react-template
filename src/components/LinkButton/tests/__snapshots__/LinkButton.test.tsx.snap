// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render component 1`] = `
<DocumentFragment>
  <a
    class="linkButton"
    href="/"
  >
    Link title
  </a>
</DocumentFragment>
`;
